src/domain/transaction/evm/evm_transaction.rs

```diff
@@ -272,6 +296,17 @@ where
         let updated_evm_data =
             updated_evm_data.with_signed_transaction_data(sig_result.into_evm()?);
 
+        let total_cost = self.calculate_total_cost(&tx).await?;
```

Comment:
This function is calling price_calculator.get_transaction_price_params again.

We call it already at line L272